services/scabbard/libscabbard/src/service/factory/factory_database.rs

```diff
@@ -0,0 +1,851 @@
+// Copyright 2018-2021 Cargill Incorporated
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::{HashMap, HashSet};
+use std::convert::TryFrom;
+use std::path::{Path, PathBuf};
+use std::sync::{Arc, Mutex, RwLock};
+use std::time::Duration;
+
+use cylinder::VerifierFactory;
+#[cfg(all(
+    feature = "receipt-store",
+    any(feature = "postgres", feature = "sqlite")
+))]
+use diesel::r2d2::{ConnectionManager, Pool};
+use openssl::hash::{hash, MessageDigest};
+#[cfg(all(
+    feature = "receipt-store",
+    any(feature = "postgres", feature = "sqlite")
+))]
+use sawtooth::receipt::store::diesel::DieselReceiptStore;
+#[cfg(feature = "receipt-store")]
+use sawtooth::receipt::store::{lmdb::LmdbReceiptStore, ReceiptStore};
+#[cfg(not(feature = "receipt-store"))]
+use sawtooth::store::{lmdb::LmdbOrderedStore, receipt_store::TransactionReceiptStore};
+use splinter::error::InvalidArgumentError;
+use splinter::error::{InternalError, InvalidStateError};
+use splinter::service::validation::ServiceArgValidator;
+use splinter::service::{FactoryCreateError, Service, ServiceFactory};
+
+use crate::hex::parse_hex;
+use crate::hex::to_hex;
+#[cfg(feature = "rest-api-actix")]
+use crate::service::rest_api::actix;
```

Comment:
Rest API code should be in its own module (actix_web_1).

Reply:
This is the same as the existing code.